monai/apps/auto3dseg/data_analyzer.py

```diff
@@ -0,0 +1,728 @@
+    which supports nii, nii.gz, png, jpg, bmp, npz, npy, and dcm formats. Currently, only segmentation
+    problem is supported, so the user needs to provide paths to the image and label files. Also, label
+    data format is preferred to be (1,H,W,D), with the label index in the first dimension. If it is in
+    onehot format, it will be converted to the preferred format.
+
+    Args:
+        datalist: a Python dictionary storing group, fold, and other information of the medical
+            image dataset, or a string to the JSON file storing the dictionary.
+        dataroot: user's local directory containing the datasets.
+        output_path: path to save the analysis result.
+        average: whether to average the statistical value across different image modalities.
+        do_ccp: apply the connected component algorithm to process the labels/images
+        device: a string specifying hardware (CUDA/CPU) utilized for the operations.
+        worker: number of workers to use for parallel processing.
+        image_key: a string that user specify for the image. The DataAnalyzer will look it up in the
+            datalist to locate the image files of the dataset.
+        label_key: a string that user specify for the label. The DataAnalyzer will look it up in the
+            datalist to locate the label files of the dataset. If label_key is None, the DataAnalyzer
+            will skip looking for labels and all label-related operations.
+
+    For example:
+
+    .. code-block:: python
+
+        from monai.apps.auto3dseg.data_analyzer import DataAnalyzer
+
+        datalist = {
+            "testing": [{"image": "image_003.nii.gz"}],
+            "training": [
+                {"fold": 0, "image": "image_001.nii.gz", "label": "label_001.nii.gz"},
+                {"fold": 0, "image": "image_002.nii.gz", "label": "label_002.nii.gz"},
+                {"fold": 1, "image": "image_001.nii.gz", "label": "label_001.nii.gz"},
+                {"fold": 1, "image": "image_004.nii.gz", "label": "label_004.nii.gz"},
+            ],
+        }
+
+        dataroot = '/datasets' # the directory where you have the image files (nii.gz)
+        DataAnalyzer(datalist, dataroot)
+
+    Notes:
+        The module can also be called from the command line interface (CLI).
+
+    For example:
+
+    .. code-block:: bash
+
+        python -m monai.apps.auto3dseg \
+            DataAnalyzer \
+            get_all_case_stats \
+            --datalist="my_datalist.json" \
+            --dataroot="my_dataroot_dir"
+
+    """
+
+    def __init__(
+        self,
+        datalist: Union[str, Dict],
+        dataroot: str = "",
+        output_path: str = "./data_stats.yaml",
+        average: bool = True,
+        do_ccp: bool = True,
+        device: Union[str, torch.device] = "cuda",
+        worker: int = 2,
+        image_key: str = "image",
+        label_key: str = "label",
```

Comment:
The type-hint should be `label_key: Optional[str]`

Reply:
Fixed